src/Core/Content/ImportExport/ImportExport.php

```diff
@@ -259,8 +260,18 @@ public function export(Context $context, ?Criteria $criteria = null, int $offset
         $this->eventDispatcher->dispatch($enrichEvent);
 
         if ($criteria->getSorting() === []) {
-            // default sorting
-            $criteria->addSorting(new FieldSorting('createdAt', FieldSorting::ASCENDING));
+            $fields = $this->repository->getDefinition()->getFields();
+            $autoIncrementFields = $fields->filterInstance(AutoIncrementField::class);
+
+            if ($autoIncrementFields->count() > 0) {
+                /** @var AutoIncrementField $autoIncrementField */
+                foreach ($autoIncrementFields as $autoIncrementField) {
+                    $criteria->addSorting(new FieldSorting($autoIncrementField->getPropertyName(), FieldSorting::ASCENDING));
```

Comment:
can you add a testcase that the export is really sorted by the auto increment if such a column exists?

Reply:
We have exports for products and orders. Both have this field. Promotions are also tested and they don't have this column. So in theory the coverage is good, but somehow these tests are not broken. An idea why I did not break the tests?

Reply:
The integration tests are not failing as any definition has these autoIncrement or an createdAt. But we need here an unit test that autoIncrement are preferred when they exist in the Criteria